### PR TITLE
feat: add Cuda support

### LINKED ISSUE #47: CUDA compilation not supported
When I specify project as mixed C / C++ / CUDA:
`project(my_project) LANGUAGES CUDA CXX C)`

the compilation of .cu files fails:
```
cd ... && /usr/local/cuda/bin/nvcc  -DPROJECT_ROOT=\"...\" -g -Xcompiler=-fPIC   -std=c++17 -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wold-style-cast -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wsign-conversion -Wnull-dereference -Wdouble-promotion -Wformat=2 -Werror -Wmisleading-indentation -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wuseless-cast -x cu -c kernels.cu -o kernels.cu.o
nvcc fatal   : Value '-Wmisleading-indentation' is not defined for option 'Werror'

### PR DESCRIPTION
**Pull request by aminya**
Fixes #47

### CODE REVIEW

**Review by DStrelak — COMMENTED** [review]
Hi, 
from my understanding, this change still does not allow for (direct) creation of a library like this:
```
add_library(my_lib)
set(sources
    alg1.cpp
    alg1.cu
    ...)
target_sources(my_lib PRIVATE ${sources})
target_link_libraries(my_lib PRIVATE project_options project_warnings)
```

File: src/Cuda.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  if(${_cuda_args_CUDA_WARNINGS} STREQUAL "")
+    target_compile_options(
+      ${target}
+      PRIVATE -Wall
+              -Wextra
+              -Wunused
+              -Wconversion
+              -Wshadow)
+  endif()
```
Comment (aminya) [inline src/Cuda.cmake:50]: I think we should remove this, and then add the Cuda warnings in `CompilerWarnings.cmake` like how @DStrelak suggested in https://github.com/cpp-best-practices/project_options/issues/47#issuecomment-1024343401

Comment (aminya) [reply src/Cuda.cmake:50]: How does your suggestion manage mixed languages? Does that add warnings for both CXX and Cuda?

File: src/CompilerWarnings.cmake [hunk 2]
```diff
@@ -69,23 +70,44 @@ function(
...
-  target_compile_options(${project_name} INTERFACE ${PROJECT_WARNINGS})
+  target_compile_options(
+    ${project_name}
+    INTERFACE $<$<COMPILE_LANG_AND_ID:CXX,Clang>:${CLANG_WARNINGS}>
+              $<$<COMPILE_LANG_AND_ID:C,Clang>:${CLANG_WARNINGS}>
+              $<$<COMPILE_LANG_AND_ID:CXX,GNU>:${GCC_WARNINGS}>
+              $<$<COMPILE_LANG_AND_ID:C,GNU>:${GCC_WARNINGS}>
+              $<$<COMPILE_LANG_AND_ID:CXX,MSVC>:${MSVC_WARNINGS}>
+              $<$<COMPILE_LANG_AND_ID:C,MSVC>:${MSVC_WARNINGS}>
+              $<$<COMPILE_LANGUAGE:Cuda>:${CUDA_WARNINGS}>)
```
Comment (aminya) [inline src/CompilerWarnings.cmake:111]: This requires some testing to ensure the warnings are enabled for C++, C, and Cuda

File: src/CompilerWarnings.cmake [hunk 3]
```diff
@@ -77,15 +89,29 @@ function(
...
+  
+  target_compile_options(
+    ${project_name}
+    INTERFACE 
+              # C++ warnings
+              $<$<COMPILE_LANGUAGE:CXX>:${PROJECT_WARNINGS_CXX}> 
+              # C warnings
+              $<$<COMPILE_LANGUAGE:C>:${PROJECT_WARNINGS_C}>
+              # Cuda warnings
+              $<$<COMPILE_LANGUAGE:CUDA>:${PROJECT_WARNINGS_CUDA}>)
```
Comment (aminya) [inline src/CompilerWarnings.cmake:115]: This requires some testing to ensure that the warnings are enabled for C++, C, and Cuda

Comment (aminya) [reply src/CompilerWarnings.cmake]: Oh, my I realized that the warnings are not being set for C++

**Comment by DStrelak** [comment]
Thanks for the changes.
At first glance it looks good, but unfortunately I do not have time to test it at the moment. I will try to do so during this week though.

**Comment by aminya** [comment]
This definitely fixes the warnings issue. I tested this on Windows with Cuda 11.6. It worked with Clang + Ninja. 

However, it failed with Visual Studio 2022 (the issue might be my project).

**Comment by aminya** [comment]
I will merge this, but I will not release it. Please test the master branch instead.

**Pull request merged**